### PR TITLE
Bug/is scripted error

### PR DESCRIPTION
**Pull request by stevedodson**
Minor mod to support 6.x style indices.

Currently, there is no specific test for this as
it requires a 6.x cluster. 6.x is not officially
supported by 7.x clients, but this is a friendly
option for users.

Longer term we should add a version check at the
high level.

### CODE REVIEW

**Comment by stevedodson** [comment]
Hopefully resolves https://github.com/elastic/eland/issues/136

File: eland/field_mappings.py [hunk 1]
```diff
@@ -162,8 +183,16 @@ def flatten(x, name=''):
         for index in mappings:
             if 'properties' in mappings[index]['mappings']:
                 properties = mappings[index]['mappings']['properties']
-
-                flatten(properties)
+            else:
```
Comment (sethmlarson) [inline eland/field_mappings.py]: Could we add a unit test so we don't regress this behavior?

File: eland/field_mappings.py [hunk 2]
```diff
@@ -162,8 +183,16 @@ def flatten(x, name=''):
...
+                # Pre Elasticsearch 7.0 mappings had types. Support these
+                # in case eland is connected to 6.x index - this is not
+                # officially supported, but does help usability
+                es_types = list(mappings[index]['mappings'].keys())
+                if len(es_types) != 1:
+                    raise NotImplementedError("eland only supports 0 or 1 Elasticsearch types")
```
Comment (sethmlarson) [inline eland/field_mappings.py]: Maybe this exception can be augmented to add the `es_types` field to give the user an indication what we're seeing on our end?

**Comment by stevedodson** [comment]
@sethmlarson - updates based on your comments.

Whilst I'm not a big fan of this change (I like strictness), I have found it very useful in practise when accessing 6.x clusters. 

We should probably add a high level check to warn if you connect to a 6.x cluster, we don't fully support or fully test.

Note, strictly the elasticsearch-py client shouldn't support 6.x clusters.

**Comment by stevedodson** [comment]
Fails CI on python 3.5 as I changed `OrderedDict` to `dict`.

**Comment by sethmlarson** [comment]
The f-string might cause an issue too, glad to see you're already in 3.6 mode :)

**Pull request merged**